tracer.go

```diff
@@ -1341,29 +1339,23 @@ func (t *Tracer) gatherMetrics(ctx context.Context, gatherers []MetricsGatherer,
 	}()
 }
 
-// canDropUnsampledTx returns true when the remote APM Server's major version is
-// >= 8, otherwise returns false.
-func (t *Tracer) canDropUnsampledTx(ctx context.Context) bool {
+// maybeDropTransaction may drop a transaction, for example when the transaction
+// is non-sampled and the target server version is 8.0 or greater.
+// maybeDropTransaction returns true if the transaction is dropped, false otherwise.
+func (t *Tracer) maybeDropTransaction(ctx context.Context, td *TransactionData, sampled bool) bool {
 	if t.versionGetter == nil {
 		return false
 	}
 
-	// Cancel the operation after 5s, we do not want to block the loop for more
-	// than necessary.
-	ctx, cancel := context.WithTimeout(ctx, 5*time.Second)
+	// Cancel the operation after 1ms, we do not want to block the loop.
+	ctx, cancel := context.WithTimeout(ctx, time.Microsecond)
 	defer cancel()
```

Comment:
This allocates, so it might be a bit expensive for hot code like the tracer loop. Can you please find a way to avoid the context.WithTimeout until we know there's no cached version?